RULES/cc/toolchain.go

```diff
@@ -10,12 +10,20 @@ import (
 
 type Toolchain interface {
 	Name() string
-	ObjectFile(out core.OutPath, depfile core.OutPath, flags []string, includes []core.Path, src core.Path) string
```

Comment:
The original idea behind this original design (that each `Toolchain` implements an interface to generate the different compile / link commands) was that this would allow DBT to support other compilers that use different flags. Your implementation now only works for compilers that act as a drop-in replacement for gcc / clang, since the commands (and thus flags) are generated by the cc rules directly.
I'm not saying that we can't go the way you propose in the PR, we should just be aware of the limitations.

Reply:
To expand a bit: The original idea was the `Toolchain` provides the generic interface and `GccToolchain` implements the interface for "gcc-like" compiilers and encapsulates the command-generation logic. So any compiler that is a drop-in replacement for gcc can become a `GccToolchain` with minimal effort, while any other compiler would require its own implementation of the `Toolchain` interface.

Reply:
And how that should work in generic way with Linked/Compiler flags in cc.Library/cc.Binary?

Reply:
You are correct, it wouldn't. Then let's fully commit to gcc-stlye compilers. If we ever need to support something else we'll have to revisit anyways.